client/app/components/ErrorBoundary.jsx

```diff
@@ -8,7 +8,10 @@ const logger = debug("redash:errors");
 
 export const ErrorBoundaryContext = React.createContext({
   handleError: error => {
-    throw error;
+    // Allow calling chain to roll up, and then throw the error in global context
```

Comment:
I found this issue when worked on ItemsList component. In short - if you throw (in this case - re-throw) an exception during some component's lifecycle events - it may completely crash app (sadly, I cannot remember details now). Using `setTimeout` allows component/React to finish whatever it needs, and when JS engine goes idle - we re-throw the error. It's completely safe because Error constructor captures all stack and when we re-throw error - we'll see correct debug info.